lfsapi/client.go

```diff
@@ -75,6 +77,41 @@ func (c *Client) Do(req *http.Request) (*http.Response, error) {
 	return res, c.handleResponse(res)
 }
 
+func (c *Client) setExtraHeaders(req *http.Request) http.Header {
+	var copy http.Header = make(http.Header)
+	for k, vs := range req.Header {
+		copy[k] = vs
+	}
+
+	for k, vs := range c.extraHeaders(req.URL) {
+		for _, v := range vs {
+			copy[k] = append(copy[k], v)
+		}
+	}
+	return copy
+}
+
+func (c *Client) extraHeaders(u *url.URL) map[string][]string {
+	if c.uc == nil {
+		c.uc = config.NewURLConfig(c.GitEnv())
```

Comment:
Good call. I had this here originally to prevent a case during testing where the `*lfsapi.Client` was initialized as `&lfsapi.Client{}` (or `new(lfsapi.Client)`) and `uc` was un-initialized. I moved this initialization to the constructor (since the value returned by `c.GitEnv()` is private and doesn't mutate), and taught `*config.URLConfig` to respond to method calls against the `nil` receiver.

x-ref: https://github.com/git-lfs/git-lfs/pull/2159/commits/fcb9fb2ba8747ee91764fa23325c089f78826c46, https://github.com/git-lfs/git-lfs/pull/2159/commits/9aa4be7d292b301b2be678347683dc4bf5535735 & https://github.com/git-lfs/git-lfs/pull/2159/commits/20cbfbabedf7876b3f5c9798c07d80e19ee1f8fa, respectively.